services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/resource_tracker_credit_transactions.py

```diff
@@ -32,11 +36,34 @@ class CreditTransactionCreateBody(BaseModel):
     created_at: datetime
 
 
+async def _sum_credit_transactions_and_publish_to_rabbitmq(
+    resource_tracker_repo: ResourceTrackerRepository,
+    credit_transaction_create_body: CreditTransactionCreateBody,
+    wallet_id: WalletID,
+    rabbitmq_client: RabbitMQClient,
+):
+    wallet_total_credits = (
+        await resource_tracker_repo.sum_credit_transactions_by_product_and_wallet(
+            credit_transaction_create_body.product_name,
+            credit_transaction_create_body.wallet_id,
+        )
```

Comment:
So after each subtraction, we recompute the sum of the credits going though all the entires in the db?
Is this correct? If Yes, could we avoid it somehow?

Reply:
My take here is, the table tracking the transactions is only going to grow. If every time we need to scan the entire thing for each user. it's only going slower over time. It's performance will degrade with size

Reply:
Yes, it is correct. This is just the "first naive version" just to get quickly going as this table is still a prerequisite to have as we need to store somewhere all the history of transactions. In the next iterations we will introduce "aggregate wallet credits table" in which we will just incrementally update the current state of the wallet so we do not need to recompute it always on the fly :+1: